mitodl/mit-open
# dashboard home

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4524

### Description (What does it do?)
This PR adds a number of `ResourceCarousel`s to the User Dashboard Home page:

 - A "top picks for you" carousel based on all of the search preferences of the current user's `Profile`: topics, certification offering and learning format
 - One for each topic in the user's `Profile`
 - A carousel for courses with / without certificates based on what the user selected in their `Profile`
 - Carousels for new and popular learning resources

### Screenshots (if appropriate):
![image](https://github.com/mitodl/mit-open/assets/12089658/3758d64f-04c1-425d-a7e2-9bceeac39eda)
![image](https://github.com/mitodl/mit-open/assets/12089658/c8dc8f3f-a1bb-42e9-9fc5-eb241dd7e89b)

### How can this be tested?
 - Spin up `mit-open` on this branch
 - Make sure you have some courses and other `LearningResource` objects in your local database by using any of the management commands for backpopulating data from the various providers
 - Make sure you are logged in and that your user has a `Profile` associated with it; this can be done either by following the onboarding process (http://localhost:8063/onboarding) or through Django Admin by creating it manually and assigning it to your user, making sure you provide some information in each section
 - Visit the site at http://localhost:8063/
 - Click your username in the header, then click "Dashboard"
 - Verify that you see all of the carousels described above, and that the courses / resources displayed within make sense in each category
 - Ensure the layout looks correct on both desktop and mobile

### Additional Context
In the design you may see some tabs above the carousels in the design. This has been deemed out of scope for now and will be addressed in the future.
